images/win/scripts/SoftwareReport/SoftwareReport.Generator.ps1

```diff
@@ -0,0 +1,196 @@
+
+$OSName = Get-OSName
+$markdown += New-MDHeader "$OSName" -Level 1
+
+$OSVersion = Get-OSVersion
+$markdown += New-MDList -Style Unordered -Lines @(
+    "$OSVersion"
+    "Image Version: $env:ImageVersion"
+)
+
+$markdown += New-MDHeader "Installed Software" -Level 2
+$markdown += New-MDHeader "Language and Runtime" -Level 3
+
+$markdown += New-MDList -Lines (Get-JavaVersionsList -DefaultVersion "1.8.0") -Style Unordered -NoNewLine
+$markdown += New-MDList -Style Unordered -Lines @(
+    (Get-RustVersion),
+    (Get-PythonVersion),
+    (Get-RubyVersion),
+    (Get-GoVersion),
+    (Get-PHPVersion),
+    (Get-JuliaVersion),
+    (Get-PerlVersion),
+    (Get-PowershellCoreVersion),
+    (Get-NodeVersion)
+)
+
+$markdown += New-MDHeader "Package Management" -Level 3
+$markdown += New-MDList -Style Unordered -Lines @(
+    (Get-ChocoVersion),
+    (Get-VcpkgVersion),
+    (Get-NPMVersion),
+    (Get-YarnVersion),
+    (Get-PipVersion),
+    (Get-CondaVersion)
+    (Get-RubyGemsVersion),
+    (Get-HelmVersion),
+    (Get-ComposerVersion)
+)
+
+$markdown += New-MDHeader "Project Management" -Level 3
+$markdown += New-MDList -Style Unordered -Lines @(
+    (Get-AntVersion),
+    (Get-MavenVersion),
+    (Get-GradleVersion)
+)
+
+$markdown += New-MDHeader "Tools" -Level 3
+$markdown += New-MDList -Style Unordered -Lines (@(
+    (Get-AzCosmosDBEmulatorVersion),
+    (Get-BazelVersion),
+    (Get-BazeliskVersion),
+    (Get-CMakeVersion),
+    (Get-DockerVersion),
+    (Get-DockerComposeVersion),
+    (Get-GitVersion),
+    (Get-GitLFSVersion),
+    (Get-InnoSetupVersion),
+    (Get-JQVersion),
+    (Get-KubectlVersion),
+    (Get-KindVersion),
+    (Get-MinGWVersion),
+    (Get-MySQLVersion),
+    (Get-MercurialVersion),
+    (Get-NSISVersion),
+    (Get-OpenSSLVersion),
+    (Get-PackerVersion),
+    (Get-SQLPSVersion),
+    (Get-SQLServerPSVersion),
+    (Get-SVNVersion),
+    (Get-WinAppDriver),
+    (Get-ZstdVersion),
+    (Get-VSWhereVersion)
+) | Where-Object { $_ })
+
+$markdown += New-MDHeader "CLI Tools" -Level 3
+$markdown += New-MDList -Style Unordered -Lines @(
+    (Get-AzureCLIVersion),
+    (Get-AzureDevopsExtVersion),
+    (Get-AWSCLIVersion),
+    (Get-AWSSAMVersion),
+    (Get-AlibabaCLIVersion),
+    (Get-CloudFoundryVersion),
+    (Get-HubVersion)
+)
+
+$markdown += New-MDHeader "Browsers and webdrivers" -Level 3
+$markdown += New-MDList -Style Unordered -Lines @(
+    (Get-BrowserVersion -Browser "chrome"),
+    (Get-SeleniumWebDriverVersion -Driver "chrome"),
+    (Get-BrowserVersion -Browser "edge"),
+    (Get-SeleniumWebDriverVersion -Driver "edge"),
+    (Get-BrowserVersion -Browser "firefox"),
+    (Get-SeleniumWebDriverVersion -Driver "firefox"),
+    (Get-SeleniumWebDriverVersion -Driver "iexplorer")
+)
+
+$markdown += New-MDHeader "MSYS2" -Level 3
+$markdown += Get-PacmanVersion
+$markdown += New-MDNewLine
+$markdown += @'
+```
+Location: C:\msys64
+
+Note: MSYS2 is pre-installed on image but not added to PATH.
+```
+'@
```

Comment:
Does it make sense to move these lines to some function to keep only markdown building in this file?